### DISCUSSION TITLE
MMKV put method in sub thread

### DISCUSSION THREAD

**Discussion by mikelhm** [Tencent/MMKV#1179] [Q&A]
MMKV putString 等方法能否放到子线程执行？ 在业务层放子线程，会不会用什么影响

火山引擎抓到一类问题堆栈发生在MMKV putString

1
at com.tencent.mmkv.MMKV.encodeString(MMKV.java:-2)
2
at com.tencent.mmkv.MMKV.putString(MMKV.java:995)

**Comment by lingol** [comment]
No.

**Reply by mikelhm** [reply]
WHy??

**Reply by lingol** [reply]
What we should do is find out why it causes ANR, because generally speaking it is **really really really fast**. Putting it in a child thread is a no-brainer so-called optimization with little thinking.
